packages/sequelize/src/index.ts

```diff
@@ -0,0 +1,156 @@
+import type { Account as ApadterAccount } from "next-auth"
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterSession,
+  VerificationToken,
+} from "next-auth/adapters"
+import { Sequelize, Model, ModelCtor } from "sequelize"
+import {
+  accountSchema,
+  userSchema,
+  sessionSchema,
+  verificationTokenSchema,
+} from "./schema"
+
+// @see https://sequelize.org/master/manual/typescript.html
+interface AccountInstance
+  extends Model<ApadterAccount, Partial<ApadterAccount>>,
+    ApadterAccount {}
+interface UserInstance
+  extends Model<AdapterUser, Partial<AdapterUser>>,
+    AdapterUser {}
+interface SessionInstance
+  extends Model<AdapterSession, Partial<AdapterSession>>,
+    AdapterSession {}
+interface VerificationTokenInstance
+  extends Model<VerificationToken, Partial<VerificationToken>>,
+    VerificationToken {}
+
+export { accountSchema, userSchema, sessionSchema, verificationTokenSchema }
+
+export default function SequelizeAdapter(client: Sequelize): Adapter {
+  const tableOptions = { underscored: true, timestamps: false }
+
+  const Account =
+    (client.models.Account as ModelCtor<any>) ||
+    client.define<AccountInstance>("account", accountSchema, tableOptions)
+  const User =
+    (client.models.User as ModelCtor<any>) ||
+    client.define<UserInstance>("user", userSchema, tableOptions)
+  const Session =
+    (client.models.Session as ModelCtor<any>) ||
+    client.define<SessionInstance>("session", sessionSchema, tableOptions)
+  const VerificationToken =
+    (client.models.VerificationToken as ModelCtor<any>) ||
+    client.define<VerificationTokenInstance>(
+      "verificationToken",
+      verificationTokenSchema,
+      tableOptions
+    )
+
+  Account.belongsTo(User, { onDelete: "cascade" })
+  Session.belongsTo(User, { onDelete: "cascade" })
+
+  return {
+    async createUser(user) {
+      return await User.create(user)
+    },
+    async getUser(id) {
+      const userInstance = await User.findByPk(id)
+
+      return userInstance?.get({ plain: true }) ?? null
```

Comment:
I love `plain: true` :heart_eyes:! This reminds me of the Prisma Adapter, which is by far the easiest to understand IMO, since it can return plain JavaScript objects, instead of convoluted custom formats for dates and stuff.

All ORMs should have an option like this! :pray:

Reply:
Agree! Having an instance returned can be useful but the option for a plain object is always great.